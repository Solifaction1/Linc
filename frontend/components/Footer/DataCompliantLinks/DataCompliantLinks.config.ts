import { Navigation } from "@models/navigation.model"

const defaultProps: Navigation = {
    audience: [],
	collapsed: false,
	createdAt: 'createdAt',
	id: 0,
	items: null,
	menuAttached: false,
	order: 0,
	parent: null,
	path: 'path',
	related: {
        createdAt: 'createdAt',
        facebook: 'facebook',
        instagram: 'instagram',
        linkedin: 'linkedin',
        locale: 'en',
        id: 1,
        navigationItemId: 1,
        publishedAt: 'publishedAt',
        sitemap_exclude: false,
        updatedAt: 'updatedAt',
        updatedBy: 'updatedBy',
        __contentType: '',
	},
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt'
}

export { defaultProps }