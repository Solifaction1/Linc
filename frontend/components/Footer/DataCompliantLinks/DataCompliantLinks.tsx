import React from 'react'
import { Navigation } from '@models/navigation.model'
import Link from '@components/Link/Link'
import LanguageSwitch from '@components/Language-switch/LanguageSwitch'
import { Localizations } from '@models/attributes.model'
import styles from './data-compliant-links.module.scss'

const DataCompliantLinks = (props: {
	className: string
	dataCompliantLinks: Navigation[]
	localizations: Localizations | undefined
}) => {
	const today = new Date()

	return (
		<div className={`${styles.dataCompliantLinks} ${props.className}`}>
			<div className={styles.dataCompliantLinksCollection}>
				<p className={styles.copyright}>
					Copyright linc.legal {today.getFullYear()}
				</p>

				<div className={styles.languageSwitch}>
					<LanguageSwitch localizations={props.localizations} />
				</div>

				<div className={styles.links}>
					{props.dataCompliantLinks?.map((dataCompliantLink) => (
						<Link
							linkType="link"
							href={`/${dataCompliantLink.path}`}
							key={dataCompliantLink.id}
							className={styles.link}
						>
							<p>{dataCompliantLink.title}</p>
						</Link>
					))}
				</div>
			</div>
		</div>
	)
}

export default DataCompliantLinks
