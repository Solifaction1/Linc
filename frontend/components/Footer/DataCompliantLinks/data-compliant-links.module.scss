@import "styles/vars";

.dataCompliantLinks {
    .dataCompliantLinksCollection {
        display: flex;
        grid-gap: 2%;
        gap: 2%;

        width: 100%;
        margin-bottom: calc(#{$spacing-base} * 2);

        .copyright, .link p {
            color: $color-white;
        }

        .copyright {
            @media (max-width: $break-md) {
                margin-bottom: calc(#{$spacing-base} * 2);
            }
            
            @media (min-width: $break-md) {
                min-width: 36rem;
                max-width: 44rem;
            }

            @media (min-width: $break-xlg) {
                width: 38rem;
            }

            margin: 0;
            font-size: $font-size-text-xxxs;
            line-height: 2.1rem;
        }
        
        @media (max-width: $break-md) {
            flex-direction: column;

            .languageSwitch {
                display: none;
            }
        }

        @media (max-width: $break-lg) {
            .languageSwitch {
                padding-top: 2rem;
            }
        }

        .languageSwitch {
            width: 35rem;

            @media (max-width: $break-xlg) {
                width: 30rem;
            }
        
            button, a {
                padding-top: 0;
                padding-left: 0;

                color: $color-white;

                line-height: 2.1rem;
                font-weight: $font-weight-regular;
            }
        }

        .link {
            p {
                background-image: none;

                font-size: $font-size-text-xxxs;
                line-height: 2.1rem;
                text-decoration: underline;
            }

            &:hover {
                text-decoration: none;

                p {
                    text-decoration: none;
                }
            }
        }
    }

    p {
        margin-right: calc(#{$spacing-base} * 1.5);

        font-size: $font-size-text-md;
        font-weight: $font-weight-regular;
        line-height: 2rem;
    }

    @media (min-width: $break-md) {
        width: 100%;

        p {
            margin-right: calc(#{$spacing-base} * 2);
        }
    }

    .links {
        display: flex;
        flex-direction: row;
        justify-content: space-between;

        @media (max-width: $break-md) {
            flex-direction: column;
            width: 100%;

            .link {
                margin-bottom: calc(#{$spacing-base} * 2);
            }
        }

        width: 32%;
    }
}