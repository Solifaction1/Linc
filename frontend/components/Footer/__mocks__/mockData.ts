/* eslint-disable max-lines */
import { Contact } from "@models/contact.model"
import { Navigation, RelatedContactInformation } from "@models/navigation.model"

const mockNavigationParent1: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 1,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent1',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 1',
    type: '',
    uiRouterKey: 'parent1',
    updatedAt: ''
}

const mockNavigationParent2: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 4,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent2',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 2',
    type: '',
    uiRouterKey: 'parent2',
    updatedAt: ''
}


export const mockContact: Contact = {
    id: 10,
    street: 'street',
    houseNumber: 'houseNumber',
    postalCode: 0,
    city: 'city',
    telephone: '+1234567890'
}

const mockContactInformationRelated: RelatedContactInformation = {
    createdAt: 'createdAt',
    facebook: 'facebook',
    id: 7,
    instagram: 'instagram',
    linkedin: 'linkedin',
    locale: 'en',
    navigationItemId: 0,
    publishedAt: 'publishedAt',
    sitemap_exclude: false,
    updatedAt: 'updatedAt',
    updatedBy: 'updatedBy',
    __contentType: '__contentType'
}

const mockContactInformation: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 8,
    items: null,
    menuAttached: true,
    order: 0,
    parent: null,
    path: 'contact-information',
    related: mockContactInformationRelated,
    sitemap_exclude: true,
    title: 'Contact Information',
    type: '',
    uiRouterKey: 'contact-information',
    updatedAt: ''
}

const mockNavigationDataCompliantLinks: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 2,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'data-compliant-links',
    related: null,
    sitemap_exclude: true,
    title: 'Data Compliant Links',
    type: '',
    uiRouterKey: 'data-compliant-links',
    updatedAt: ''
}

export const mockNavigationData = [
    mockNavigationParent1,
    mockNavigationParent2,
    mockContactInformation,
    mockNavigationDataCompliantLinks,
]