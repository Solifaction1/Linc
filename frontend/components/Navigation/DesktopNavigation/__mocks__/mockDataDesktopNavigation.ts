/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    items: 'items',
    parent: null,
    related: null
}]
