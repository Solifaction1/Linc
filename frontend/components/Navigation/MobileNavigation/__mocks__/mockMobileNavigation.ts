/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    children: [],
    items: 'items',
    parent: null,
    related: {
        createdAt: '',
        facebook: 'https://www.facebook.com',
        linkedin: 'https://www.linkedin.com',
        instagram: '',
        locale: '',
        id: 1,
        navigationItemId: 2,
        publishedAt: '',
        sitemap_exclude: false,
        updatedAt: '',
        updatedBy: '',
        __contentType: ''
    }
}]
