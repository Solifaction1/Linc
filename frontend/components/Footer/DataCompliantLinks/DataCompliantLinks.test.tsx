import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import DataCompliantLinks from './DataCompliantLinks'
import { mockNavigationData } from '../__mocks__/mockData'

jest.mock('next/router', () => ({
	useRouter() {
		return {
			locale: 'locale',
			pathname: 'test/test',
		}
	},
}))

describe('DataCompliantLinks', () => {
	it('renders the DataCompliantLinks', () => {
		const { container, getByText } = render(
			<DataCompliantLinks
				className="className"
				dataCompliantLinks={mockNavigationData}
				localizations={{ data: undefined }}
			/>
		)

		expect(container).toMatchSnapshot()

		expect(
			container.getElementsByClassName('dataCompliantLinks className')[0]
		).toBeVisible()

		expect(
			container.getElementsByClassName('dataCompliantLinksCollection')[0]
		).toBeVisible()

		expect(container.getElementsByClassName('copyright')[0]).toBeVisible()
		expect(
			getByText('Copyright linc.legal 2023', { selector: 'p' })
		).toBeVisible()

		expect(container.getElementsByClassName('languageSwitch')[0]).toBeVisible()
		expect(container.getElementsByClassName('languageSwitch')[1]).toBeVisible()

		expect(
			container.getElementsByClassName('languageLink activeLang')[0]
		).toBeVisible()
		expect(getByText('LOCALE', { selector: 'button' })).toBeVisible()

		expect(container.getElementsByClassName('links')[0]).toBeVisible()

		expect(
			container.getElementsByClassName('link link link md false')[0]
		).toBeVisible()
		expect(container.querySelector('a[href="/parent1"]')).toBeVisible()

		expect(
			container.getElementsByClassName('link link link md false')[1]
		).toBeVisible()
		expect(container.querySelector('a[href="/parent2"]')).toBeVisible()

		expect(
			container.getElementsByClassName('link link link md false')[2]
		).toBeVisible()
		expect(
			container.querySelector('a[href="/contact-information"]')
		).toBeVisible()
		expect(getByText('Contact Information', { selector: 'p' })).toBeVisible()

		expect(
			container.getElementsByClassName('link link link md false')[3]
		).toBeVisible()
		expect(
			container.querySelector('a[href="/data-compliant-links"]')
		).toBeVisible()
		expect(getByText('Data Compliant Links', { selector: 'p' })).toBeVisible()
	})
})
