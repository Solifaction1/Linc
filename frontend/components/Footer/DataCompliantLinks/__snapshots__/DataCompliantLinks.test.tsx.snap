// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataCompliantLinks renders the DataCompliantLinks 1`] = `
<div>
  <div
    class="dataCompliantLinks className"
  >
    <div
      class="dataCompliantLinksCollection"
    >
      <p
        class="copyright"
      >
        Copyright linc.legal 
        2023
      </p>
      <div
        class="languageSwitch"
      >
        <div
          class="languageSwitch"
        >
          <button
            class="languageLink activeLang"
            type="button"
          >
            LOCALE
            <svg
              class="false arrow"
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
              />
            </svg>
          </button>
          <div
            class="accordionWrapper"
            style="height: 0px;"
          >
            <div
              class="languageLinkWrapper"
            />
          </div>
        </div>
      </div>
      <div
        class="links"
      >
        <a
          class="link link link md false"
          href="/parent1"
        >
          <p>
            Parent 1
          </p>
        </a>
        <a
          class="link link link md false"
          href="/parent2"
        >
          <p>
            Parent 2
          </p>
        </a>
        <a
          class="link link link md false"
          href="/contact-information"
        >
          <p>
            Contact Information
          </p>
        </a>
        <a
          class="link link link md false"
          href="/data-compliant-links"
        >
          <p>
            Data Compliant Links
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
